tests/jobs/pyscal/test_pyscal.py

```diff
@@ -0,0 +1,205 @@
+
+
+EXAMPLE_STATIC_DFRAME = pd.DataFrame(
+    columns=["SATNUM", "Nw", "Now", "Nog", "Ng"],
+    data=[[1, 2, 2, 3, 3], [2, 4, 4, 1, 1]],
+)
+EXAMPLE_SCAL = pd.DataFrame(
+    columns=["SATNUM", "CASE", "Nw", "Now", "Nog", "Ng"],
+    data=[[1, "Low", 2, 2, 3, 3], [1, "Base", 4, 4, 1, 1], [1, "High", 5, 5, 2, 2]],
+)
+
+
+def test_fm_pyscal_static_csv(tmpdir):
+    """Test fm_pyscal on a static relperm input as CSV"""
+    tmpdir.chdir()
+    EXAMPLE_STATIC_DFRAME.to_csv("relperm-input.csv", index=False)
+    run(
+        "relperm-input.csv", "relperm.inc", "", "__NONE__", "__NONE__", "sgof", 1,
+    )
+    assert os.path.exists("relperm.inc")
+
+
+def test_fm_pysal_static_xlsx(tmpdir):
+    """Test fm_pyscal on a static xlsx file"""
+    tmpdir.chdir()
+    EXAMPLE_STATIC_DFRAME.to_excel("relperm-input.xlsx")
+    run(
+        "relperm-input.xlsx", "relperm.inc", "", "__NONE__", "__NONE__", "slgof", 1,
+    )
+    assert os.path.exists("relperm.inc")
+
+
+def test_fm_pyscal_scal(tmpdir):
+    """Test interpolation in a SCAL recommendation"""
+    tmpdir.chdir()
+    with open("parameters.txt", "w") as p_file:
+        p_file.write("INTERPOLATE_WO 0.1\nINTERPOLATE_GO 0.5")
+    EXAMPLE_SCAL.to_csv("scal-input.csv", index=False)
+    run(
+        "scal-input.csv",
+        "relperm.inc",
+        "",
+        "INTERPOLATE_WO",
+        "INTERPOLATE_GO",
+        "sgof",
+        1,
+    )
+    assert os.path.exists("relperm.inc")
+
+    # Test with same interpolation parameter for both:
+    run(
+        "scal-input.csv",
+        "relperm-wo-wo.inc",
+        "",
+        "INTERPOLATE_WO",
+        "INTERPOLATE_WO",
+        "sgof",
+        1,
+    )
+    assert os.path.exists("relperm-wo-wo.inc")
+    run(
+        "scal-input.csv",
+        "relperm-wo-none.inc",
+        "",
+        "INTERPOLATE_WO",
+        "__NONE__",
+        "sgof",
+        1,
+    )
+    assert os.path.exists("relperm-wo-none.inc")
+    # These files should be identical
+    wowolines = "".join(open("relperm-wo-wo.inc").readlines())
+    wononelines = "".join(open("relperm-wo-none.inc").readlines())
+    assert wowolines == wononelines
+    # (this could fail if pyscal later injects random characters in say
+    # its comments, if so, remove all lines starting with "--")
+
+
+def test_fm_pyscal_scal_genkw(tmpdir):
+    """Test interpolation in a SCAL recommendation with GEN_KW parameters
+
+    GEN_KW will add a prefix before a colon - we choose to ignore that
+    prefix altogether in FM_PYSCAL"""
+
+    tmpdir.chdir()
+    with open("parameters.txt", "w") as p_file:
+        p_file.write("INTERPOLATE_WO 0.1\nRELPERM:INTERPOLATE_GO 0.5")
+    EXAMPLE_SCAL.to_csv("scal-input.csv", index=False)
+    run(
+        "scal-input.csv",
+        "relperm.inc",
+        "",
+        "RELPERM:INTERPOLATE_WO",
+        "INTERPOLATE_GO",
+        "sgof",
+        1,
+    )
+    assert os.path.exists("relperm.inc")
+
+
+def test_fm_pyscal_errors(tmpdir):
```

Comment:
These tests should be parametrized, meaning that the test is run multiple times with varying input. The varying input here would be e.g. `expected_exceptions`, `run_args`, `static_dframe`, `indexed_csv`, etc.

Comments on the assertions/setups can be in the parameter data.

Reply:
Done.